src/main/groovy/org/gradle/android/workarounds/MergeNativeLibsWorkaround.groovy

```diff
@@ -0,0 +1,34 @@
+package org.gradle.android.workarounds
+
+import com.android.build.gradle.internal.tasks.MergeNativeLibsTask
+import org.gradle.android.AndroidIssue
+import org.gradle.api.Task
+import org.gradle.api.file.FileCollection
+
+import java.lang.reflect.Field
+
+/**
+ * In 3.5.x, projectNativeLibs is not directly backed by state in the Task object.  Instead, it is effectively a
+ * method that calculates a new FileCollection from variantScope every time it is called.  So we only
+ * support this workaround for 3.6.x and 4.0.x.
+ */
+@AndroidIssue(introducedIn = "3.6.0", fixedIn = ["4.1.0"], link = "https://issuetracker.google.com/issues/140602655")
+class MergeNativeLibsWorkaround extends AbstractAbsolutePathWorkaround {
```

Comment:
I think the real issue with MergeNativeLibsWorkaround is that it shouldn't be cacheable at all, it produces a big output that is more expensive to download than to execute (even when you have not a single NativeLib on the project).
Scans say you avoid some time using the build cache but it's only because the time used for the first build includes gzip and upload.

Reply:
Please mention these concerns on the Android plugin ticket: https://issuetracker.google.com/issues/140602655